httpcore/_async/http_proxy.py

```diff
@@ -172,33 +172,22 @@ async def _forward_request(
 
         return status_code, headers, wrapped_stream, ext
 
-    async def _tunnel_request(
```

Comment:
Great stuff! So...

I think the first thing would be to try to get the change footprint of this PR down a bit.

* I'd suggest we *keep* a single `_tunnel_request` method, rather than splitting out into two methods here?
* Can we catch the exception more narrowly? Perhaps we could have a top-level `TransportError` class, in the same way as `httpx`. Then, in cases like this we could *just* catch `TransportError`.

Reply:
Actually thinking about it let's treat that second line item *independently* of this pull request.

Reply:
Do I understand correctly the plan:
1. to introduce the root exception for all `httpcore` exceptions, to name the class `TransportError`. This change should go in the separate PR
2. as soon as (1) has been merged, to merge `_establish_tunnel_proxy_connection` back into `_tunnel_request`, catching `TransportError` instead of `Exception`?

Talking about (2) we can try. But also it means that there are cases (some error conditions) when we do not clean up resources. The conditions are exceptions not inherited by `TransportError`...

Reply:
No don't worry about (2) at all. We can think about that some other day.

Reply:
Therefore I'm to just eliminate `_establish_tunnel_proxy_connection`, moving this code back to `_tunnel_request` to reduce the PR footprint, yes? _(sorry for repeating questions, I want to fully understand you)_

Reply:
@tomchristie 
I did that, so the PR became significantly smaller (but it's required to compare it ignoring whitespace changes)